src/python/dxpy/bindings/dxapplet.py

```diff
@@ -58,11 +58,13 @@ def _get_run_input_common_fields(executable_input, **kwargs):
             if kwargs.get(arg) is not None:
                 run_input[arg] = kwargs[arg]
 
-        if kwargs.get('instance_type') is not None or kwargs.get('cluster_spec') is not None or kwargs.get('fpga_driver') is not None:
+        if (kwargs.get('instance_type') is not None or kwargs.get('cluster_spec') is not None or
```

Comment:
We could use 
`if any(kwargs.get(key) for key in ['instance_type', 'cluster_spec', 'fpga_driver', 'nvidia_driver']):`

(note that this checks for "truthy values", not just Nones, previously the value "" would pass the check, now it wouldnt, but that should not really matter)

alternatively : `if any(kwargs.get(key) is not None for key in ['instance_type', 'cluster_spec', 'fpga_driver', 'nvidia_driver']):`

Reply:
Looks much better - fixed. I used the format with 'is not None' to be sure we dont change behaviour for any possible case.